### PR TITLE
Done setup and custom theme-provider (#5)

### PR DESCRIPTION
**Pull request by dinhthi12**
# Related Tickets

- [#Ticket_ID](https://edu-redmine.vn/issues/xxxx)

## WHAT this PR does?

- Pull setup project and custom theme.

## Notes Impacted Areas

- N/A.

## Attachments

- N/A.

### CODE REVIEW

File: app/template.jsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ * Dòng 19-31: Định nghĩa component Template.
+ * Dòng 20-26: Sử dụng motion.main để bao bọc nội dung chính của ứng dụng với animation. variants được sử dụng để xác định các biến thể của animation, initial và animate để thiết lập trạng thái ban đầu và trạng thái hiện tại của animation, transition để thiết lập hiệu ứng chuyển động của animation.
+ * Dòng 29-30: Sử dụng useScrollUp() để hiển thị nút cuộn lên khi cần thiết.
+ * Dòng 31: Sử dụng một span để tạo thanh tiến độ cuộn trang, sử dụng giá trị của completion (trả về từ hook useScrollProgress) để điều chỉnh vị trí và chiều dài của thanh tiến độ.
+ */
+const Template = ({ children }) => {
+  const completion = useScrollProgress();
+  return (
+    <>
+      <motion.main
```
Comment (tn0810520) [inline app/template.jsx:28]: Ok, nên sửa lại chi tiết cmt hơn để bữa sau đọc dễ hiểu

File: hooks/useScrollProgress.jsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import React, { useState, useEffect } from "react";
+
+const useScrollProgress = () => {
```
Comment (tn0810520) [inline hooks/useScrollProgress.jsx:3]: Hooks ok

File: hooks/useScrollUp.jsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+   * Dòng 23-25: Loại bỏ sự kiện lắng nghe scroll khi component unmounts để tránh memory leaks.
+   */
+  const [isVisible, setIsVisible] = useState(true);
+
+  useEffect(() => {
+    const handleScroll = () => {
+      const currentScrollPos = window.pageYOffset;
+      const isVisible = currentScrollPos > window.innerHeight / 2; // Hiển thị nút khi cuộn xuống dưới một nửa chiều cao của trang
+      setIsVisible(isVisible);
+    };
```
Comment (tn0810520) [inline hooks/useScrollUp.jsx:21]: Tốt 10đ kkkk

**Comment by hoangpt1310** [comment]
Merged

**Pull request merged**